ADR/nnnn-revised-component-model.md

```diff
@@ -0,0 +1,59 @@
+# NN. Revised Component Model
+
+## Status
+
+Proposed
+[//]: # (One of: Proposed, Withdrawn, Deferred, Accepted, Implementable, Implemented, Replaced)
+
+## Context
+
+The existing application and component model in Konflux presents several
+limitations that hinder our ability to scale and support flexible workflows.
+Specifically, the current model requires a new component to be registered for
+each use case, even when the same underlying code is being used in multiple
+release scenarios. This creates unnecessary overhead and complexity, while also
+limiting the reusability of our components.
+
+Furthermore, ownership of the current APIs is unclear. They were originally
+owned by HAS (which is now deprecated) and exhibit excessive coupling across
+multiple services.
+
```

Comment:
What would be the changes to the annotations and labels the konflux controllers set on the pipelinerun definition? Those play crucial role and affect the steps that should be taken when a new branch is created and have the pipelinerun definition of the previous branch.

Reply:
I think we may not have that level of detail yet. @mmorhun - have you thought about this enough yet to comment on this one?